File: Algorithm/CPP/summary_ranges.cpp
# include <vector>
# include <algorithm>
# include <string>
# include <stack>
# include <algorithm>
# include <deque>
# include <iostream>
using namespace std;




class Solution {
public:
	string range2str(int start, int end)
	{
		string res;
		res += to_string(start);
		if (start == end)		return res;
		res += "->";
		res += to_string(end);
		return res;		
	}
	vector<string> summaryRanges(vector<int>& nums) {
		int n = nums.size();
		vector<string> res;
		if (n == 0) return res;
		int start = nums[0];
		int pre = nums[0];
		
		for (int i = 1; i < n; i++)
		{
			if (nums[i] > pre + 1)
			{
				res.push_back(range2str(start, pre));
				start = nums[i];
			}
			pre = nums[i];
		}
		res.push_back(range2str(start, pre));
		return res;
	}
};

int main()
{
	Solution sol;
	vector<int> nums = {-1};

	for(auto k:sol.summaryRanges(nums)) cout<<k<<" ";

	cout << endl;
	//for (auto k :nums )cout << k << " ";
	system("pause");
	return 0;
}

File: Algorithm/Python/summary_ranges.py
class Solution:
    # @param {integer[]} nums
    # @return {string[]}
    def summaryRanges(self, nums):
        def getRange(begin, end):
            return str(begin)+"->" + str(end) if begin != end else str(begin)
        n = len(nums)
        if n == 0:
            return []
        res = []
        pre = start = nums[0]
        for num in nums:
            cur = num 
            if cur - pre > 1:
                res.append(getRange(start, pre))
                start = cur
            if cur == nums[-1]:
                res.append(getRange(start, cur))
            pre = cur
        return res
                
        



if __name__ == '__main__':
    nums = [0,1,2,3,4,5,6]
    sol = Solution()
    print sol.summaryRanges(nums)
